### ISSUE TITLE
Scopes and findAndCountAll wrong count query issue

### ISSUE THREAD

**Issue by echmykhun** [sequelize/sequelize#6235]
Hello!

During using sequelize, i faced following issue: 
If i'm using `scopes` and `findAndCountAll()`, result count will be wrong.

For example, if i have 10 products and several 1:n or n:m scopes, let it be assets(1:n) and categories(n:m). I have several assets and categories for each product
Product model part
```js
//...
scopes:{
            assets: function () {
                return {
                    include: [{association: product.associations.assets}]
                };
            },
            categories: function () {
                return {
                    include: [{association: product.associations.categories}]
                };
            }
 }
//...
```
```js
Product.scope('categories', 'assets').findAndCountAll()
.then(result => { /* code */ })
```
Here, as result.count, i expect `10`, since i have 10 products, but i got 321, since it calls COUNT on query with JOIN's and i have several assets and several categories for each product.
Generated sql for count:
```SQL
SELECT count(`Product`.`id`) AS `count` FROM `products` AS `Product` LEFT OUTER JOIN `product_assets` AS `assets` ON `Product`.`id` = `assets`.`product_id` LEFT OUTER JOIN (`product_categories` AS `categories.ProductCategory` INNER JOIN `categories` AS `categories` ON `categories`.`id` = `categories.ProductCategory`.`category_id`) ON `Product`.`id` = `categories.ProductCategory`.`product_id`
```

If i use simple `include` like this:
```js
Product.findAndCountAll({
                include: [
                    this.model.associations.assets,
                    this.model.associations.categories
                ]
            })
```
It works as expected

__Dialect:__ mysql 
__Sequelize:__ version: 3.23.4

**Comment by sushantdhiman** [comment]
I think https://github.com/sequelize/sequelize/pull/6190 solves it. It will be published soon

**Comment by BridgeAR** [comment]
You have to use the `distinct` option in this case.

**Comment by echmykhun** [comment]
@sushantdhiman Hi!

With sequelize version 3.23.6 I faced the same behavior, even when i use simple includes. When write something like this: 
```js
Product.findAndCountAll({
                include: [
                    this.model.associations.assets
                ]
            })
```

It runs count query with ```LEFT JOIN```
Also, scopes don't do the thing as well

**Comment by tolgaek** [comment]
@BridgeAR what do you mean by the distinct option? I have the same issue where I have 1:m relationships and I think the count is returning the total count vs the distinct count

**Comment by echmykhun** [comment]
@BridgeAR, I got it working with this:
```javascript
Product.scope('categories', 'assets').findAndCountAll({distinct: 'product.id'})
.then(result => { /* code */ })
```

**Comment by BridgeAR** [comment]
@tolgaek you have to set `distinct: true`. Currently the docs miss the hint that not only the findAll options but also the clount options can and should be passed to the function.

@echmykhun you actually should set it to `true` instead of using `product.id` that evaluates to true.

**Comment by ippeiukai** [comment]
Hi @sushantdhiman,

We were bitten by the (presumably) same issue on our first attempt to update sequelize from 3.23.3 to 3.23.6 (https://github.com/sequelize/sequelize/compare/v3.23.3...v3.23.6). We have reverted back with regression failures and trying to figure out how best to deal with this change.

We have a complicated eager loading for our models and used `findAndCountAll()` to get the count of models along side `limit`ed results (we don't care how many rows there may be while eager loading). With the update the behaviour of `findAndCountAll()` has changed to return inflated number of `count`s.

Is the spec now such that in our use case we should make separate calls of `count()` and `findAll()` with slightly different query?

**Comment by ippeiukai** [comment]
@sushantdhiman
> We were bitten by the (presumably) same issue

Sorry, but I think we figured out what was happening. 
Basically, we had some includes with `where` but with no `required: false` specified (those associations that have always at least one record present, basically). So the issue wasn't the same as this one, and our code was wrong.

**Comment by netdragonboberb** [comment]
distinct solved it for us with 3.23.4, but for some reason VS Code highlights the line and says distinct isn't in FindOptions. Either way, it works.

**Comment by felixfbecker** [comment]
@netdragonboberb If TypeScript complains, please open an issue at https://github.com/types/npm-sequelize
